manifests/repository.pp

```diff
@@ -19,12 +19,12 @@
 #   The major bareos release version which should be used
 #
 class bareos::repository (
-  Enum['18.2', '19.2', '20', '21']  $release             = '20',
+  Enum['18.2', '19.2', '20', '21']  $release             = $bareos::params::repo_release,
   Optional[String[1]]               $gpg_key_fingerprint = undef,
   Boolean                           $subscription        = false,
   Optional[String]                  $username            = undef,
   Optional[String]                  $password            = undef,
-) {
+) inherits bareos::params {
```

Comment:
I'm wondering if that's required. Do we allow people to use this class directly or should it be marked private? the init.pp calls bareos::repository, that means `$bareos::repo_release` should work.

Reply:
I think we can continue to allow people to use this class directly.

Reply:
in that case we maybe should inherit from bareos instead? that would be a common pattern. Otherwise you end up in situations where people set bareos::repo_release in hiera but that won't be picked up by this class.

Reply:
Then we can use this default parameter (no `params` component):

```puppet
    Enum['18.2', '19.2', '20', '21']  $release             = $bareos::repo_release,
```

No need for inheriting, we can even remove the parameter and use the `$bareos::repo_release` variable directly, and make the class private without impacting end-users :wink:

Reply:
I'm sorry but I don't understand what is the advantage of making the `param` class private.
What is the problem to allow user to only add Bareos repository and nothing else?

Reply:
I am not speaking about the `params` class sorry… As you said, `bareos::repository` cannot be declared by the end-user because the `bareos` class declares it. So the user has no possibility to pass custom parameters unless using hiera which is not a good idea, it is some implementation detail internal to the module.

Reply:
Ooohh no, my bad, I write `param `instead of `repository `. I'm sorry.
This scenario is true only if `bareos::repository ` inherit of `bareos` but we have no problem if  `bareos::repository ` inherit of `bareos::param` as I wrote in code.

Reply:
I believe that @smortex is suggesting to reference the `repo_release` parameter of the `bareos` class and no inheritance is needed.